components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -115,6 +122,56 @@ public ApplicationModel getApplication(String applicationId) {
         }
     }
 
+    /**
+     * Export an application identified by the applicationId, as an XML string.
+     *
+     * @param applicationId ID of the application to be exported.
+     * @param exportSecrets If True, all hashed or encrypted secrets will also be exported.
+     * @return XML string of the application.
+     */
+    public String exportApplication(String applicationId, Boolean exportSecrets) {
+
+        try {
+            // Pull application using the ID to get the App name
+            String tenantDomain = ContextLoader.getTenantDomainFromContext();
+            ApplicationBasicInfo application =
+                    getApplicationManagementService().getApplicationBasicInfoByResourceId(applicationId, tenantDomain);
+            if (application == null) {
+                throw buildApiError(ErrorMessage.ERROR_CODE_APPLICATION_NOT_FOUND, applicationId, tenantDomain);
+            }
+            String appName = application.getApplicationName();
+            // Pass app name and get the XML output
+            return getApplicationManagementService().exportSPApplication(appName, exportSecrets, tenantDomain);
```

Comment:
Shall we introduce a method at OSGi level to export SP using resourceId?

Reply:
Added with https://github.com/wso2/carbon-identity-framework/pull/2534